addons/web/static/src/js/views/form_relational_widgets.js

```diff
@@ -954,24 +954,26 @@ var X2ManyListView = ListView.extend({
         if (_.isEmpty(this.records.records)){
             return true;
         }
+        var fields = this.editor.form.fields;
         var current_values = {};
-        _.each(this.editor.form.fields, function(field){
+        _.each(fields, function(field){
             field._inhibit_on_change_flag = true;
             field.no_rerender = true;
             current_values[field.name] = field.get('value');
         });
-        var valid = _.every(this.records.records, function(record){
-            _.each(self.editor.form.fields, function(field){
-                field.set_value(record.attributes[field.name]);
+        var valid = _.every(this.dataset.cache, function(record){
+            _.each(fields, function(field){
+                var value = record.values[field.name];
+                field.set_value(_.isArray(value) && _.isArray(value[0]) ? [[5]].concat(value) : value);
```

Comment:
maybe comment like this:
// if nested commands, removes all records from set first